src/scipp/compat/pandas_compat.py

```diff
@@ -3,53 +3,108 @@
+    )
+
+    coords = (
+        {row_index_name: array(dims=[row_index_name], values=row_index)}
+        if include_index
+        else {}
+    )
 
+    if se.dtype == "string":
+        # se.to_numpy() and np.array(se.values) produce an array of dtype=object
+        # when the series contains strings.
+        values = se.to_numpy(dtype=str)
+    else:
+        values = se.to_numpy()
     return DataArray(
-        data=Variable(values=se.values, dims=[row_index_name]),
-        coords={row_index_name: Variable(dims=[row_index_name], values=row_index)},
+        data=array(values=values, dims=[row_index_name], unit=unit),
+        coords=coords,
         name=name,
     )
 
 
-def from_pandas_dataframe(df: pd.DataFrame) -> Dataset:
+def from_pandas_dataframe(
+    df: pd.DataFrame,
+    *,
+    data_columns: Optional[Union[str, Iterable[str]]] = None,
+    include_index: bool = True,
+    header_parser: HeaderParserArg = None,
+) -> Dataset:
     import pandas as pd
 
-    row_index = df.axes[0]
-    row_index_name = row_index.name or "row"
-
-    if df.ndim == 1:
-        # Special case for 1d dataframes, treat them as a series, but still
-        # wrap them in a dataset object for consistency of return types.
-        return Dataset(data={row_index_name: from_pandas_series(pd.Series(df))})
+    columns = (
+        from_pandas_series(
+            pd.Series(df[column_name]),
+            include_index=include_index,
+            header_parser=header_parser,
+        )
+        for column_name in df.axes[1]
+    )
+    coords = {da.name: da for da in columns}
 
-    sc_data = {}
-    for column_name in df.axes[1]:
-        sc_data[f"{column_name}"] = from_pandas_series(pd.Series(df[column_name]))
+    if data_columns is None:
+        data = coords
+        coords = {}
+    else:
+        if isinstance(data_columns, str):
+            data_columns = (data_columns,)
+        data = {name: coords.pop(name) for name in data_columns}
+        coords = {name: coord.data for name, coord in coords.items()}
 
-    return Dataset(data=sc_data)
+    return Dataset(data, coords=coords)
 
 
-def from_pandas(pd_obj: Union[pd.DataFrame, pd.Series]) -> VariableLike:
+def from_pandas(
+    pd_obj: Union[pd.DataFrame, pd.Series],
+    *,
+    data_columns: Optional[Union[str, Iterable[str]]] = None,
+    include_index: bool = True,
+    header_parser: HeaderParserArg = None,
+) -> VariableLike:
     """Converts a pandas.DataFrame or pandas.Series object into a
     scipp Dataset or DataArray respectively.
 
     Parameters
     ----------
     pd_obj:
         The Dataframe or Series to convert.
+    data_columns:
+        Select which columns to assign as data.
+        The rest are returned as coordinates.
+        If ``None``, all columns are assigned as data.
+        Use an empty list to assign all columns as coordinates.
+    include_index:
+        If True, the row index is included in the output as a coordinate.
```

Comment:
I wonder if we could by defaulr exclude the `RangeIndex` that you get automatically from Pandas?

```python
import pandas as pd
df = pd.DataFrame({'a':[1,2,3,4]})
df.index  # RangeIndex(start=0, stop=4, step=1)
```
but include all others?

Reply:
Why? This would make the interface more complicated.

Reply:
Because the `RangeIndex` probably exists only for technical reasons and is otherwise meaningless?